### PR TITLE
Move nodeunit into devDependencies.

### PR DESCRIPTION
**Pull request by jsirois**
Hello @dpup, @dudeche, 

Please review the following commits I made in branch 'jsirois/draccus/fix_deps'.

f5213a16510310823ad3c1134b194fbc5c0476f6 (2015-08-25 13:27:31 -0600)
Move nodeunit into devDependencies.
This also gets Travis testing against node 0.12 and using the new
container infrastructure.

R=@dpup
R=@dudeche

MANUAL TESTING=rm -rf node_modules/ && npm install && npm test

Found this exploring draccus as my sample project for pants node
support as suggested by @huckphin.  I selected you as reviewers
via git log stats.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,11 @@
...
+# which should provide faster startup times and beefier
+# "machines".
+# See: http://docs.travis-ci.com/user/workers/container-based-infrastructure/
+sudo: false
+
 language: node_js
 node_js:
+  - "0.12"
   - "0.10"
   - "0.8"
```
Comment (jsirois) [inline .travis.yml]: Hmm node 0.8 won't setup on the Travis boxes.  I notice, for example, [kew](https://github.com/Medium/kew/blob/master/.travis.yml) doesn't try to address 0.8.  Can I drop this fro the matrix?

Comment (nicks) [inline .travis.yml]: ya, node 0.8 is EOL'd, just drop it

Comment (jsirois) [inline .travis.yml]: Excellent - ty.  Done.

**Comment by jsirois** [comment]
@nicks - I spoke with @dudeche in #mono and we agree we're both draccus noobs and @dpup is a busy guy.  It looks like you are a very busy guy too based on github stats but you did respond - so can you LGTM/fix x/y/z for this review?

File: .travis.yml [hunk 2]
```diff
@@ -1,4 +1,10 @@
+# Enables support for a docker container-based build
+# which should provide faster startup times and beefier
+# "machines".
+# See: http://docs.travis-ci.com/user/workers/container-based-infrastructure/
+sudo: false
```
Comment (nicks) [inline .travis.yml]: cool!

**Comment by nicks** [comment]
lgtm

**Comment by jsirois** [comment]
Thanks @nicks !

**Pull request merged**